src/io/flutter/survey/FlutterSurveyNotifications.java

```diff
@@ -0,0 +1,158 @@
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.fileEditor.FileEditorManagerEvent;
+import com.intellij.openapi.fileEditor.FileEditorManagerListener;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.jetbrains.lang.dart.DartFileType;
+import icons.FlutterIcons;
+import io.flutter.FlutterInitializer;
+import io.flutter.FlutterMessages;
+import io.flutter.pub.PubRoot;
+import org.jetbrains.annotations.NotNull;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+// TODO(pq): remove for M36 / post: 2019/05/27.
+public class FlutterSurveyNotifications {
+  private static final int NOTIFICATION_DELAY_IN_SECS = 3;
+
+  private static final String FLUTTER_LAST_SURVEY_CHECK_KEY = "FLUTTER_LAST_SURVEY_CHECK_KEY";
+  private static final long CHECK_INTERVAL_IN_MS = TimeUnit.HOURS.toMillis(40);
+
+  private static final String SURVEY_TAKEN = "io.flutter.survey.2019.alreadyTaken";
+  private static final String SURVEY_URL = "https://google.qualtrics.com/jfe/form/SV_3W3aVD2y9CoAe6V?Source=IntelliJ";
+
+  private static long SURVEY_START;
+  private static long SURVEY_END;
+
+  static {
+    final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm");
+    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    try {
+      SURVEY_START = dateFormat.parse("2019/05/13 19:00").getTime();
+      SURVEY_END = dateFormat.parse("2019/05/27 07:00").getTime();
+    }
+    catch (ParseException e) {
+      // Shouldn't happen.
+    }
+  }
+
+  private static boolean isSurveyOpen() {
+    final long now = System.currentTimeMillis();
+    return now >= SURVEY_START && now <= SURVEY_END;
+  }
+
+  interface FlutterSurveyNotifier {
+    void prompt();
+  }
+
+  public static void init(@NotNull Project project) {
+    project.getMessageBus().connect().subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, new FileEditorManagerListener() {
+      @Override
+      public void fileOpened(@NotNull final FileEditorManager source, @NotNull final VirtualFile file) {
+        check(file);
+      }
+
+      @Override
+      public void selectionChanged(@NotNull FileEditorManagerEvent event) {
+        final VirtualFile file = event.getNewFile();
+        if (file != null) {
+          check(file);
+        }
+      }
+
+      private void check(@NotNull VirtualFile file) {
+        if (PubRoot.isPubspec(file) || file.getFileType() == DartFileType.INSTANCE) {
+          new FlutterSurveyNotifications(project).checkForDisplaySurvey();
+        }
+      }
+    });
+  }
+
+  @NotNull final Project myProject;
```

Comment:
If I open two projects will I get notified twice?

Reply:
Nope.   That state is cached in a global property.  (Thanks for asking!)